src/sp_execute.c

```diff
@@ -102,15 +102,16 @@ is_in_eval_and_whitelisted(const zend_execute_data *execute_data) {
 /* This function gets the filename in which `eval()` is called from,
  * since it looks like "foo.php(1) : eval()'d code", so we're starting
  * from the end of the string until the second closing parenthesis. */
-char *get_eval_filename(const char *const filename) {
-  size_t i = strlen(filename);
+zend_string *get_eval_filename(const char *const filename) {
   int count = 0;
-  char *clean_filename = estrdup(filename);
+  zend_string *clean_filename = zend_string_init(filename, strlen(filename), 0);
 
-  while (i--) {
-    if (clean_filename[i] == '(') {
+  // TODO maight cause some problem, might need to adjust length
```

Comment:
Yes please :<

Reply:
Is already fixed, forgot to remove the TODO.